src/main/java/objects/inventory/InventoryState.java

```diff
@@ -0,0 +1,151 @@
+package objects.inventory;
+
+import interfaces.InputValidator;
+import interfaces.State;
+import objects.item.*;
+
+/**
+ * Use case to change inventory based on user input
+ */
+public class InventoryState implements State {
+    /**
+     * inventory: entity for inventory
+     * isDone: whether the state is done
+     * awaitingInput: whether the state is waiting for user input
+     * viewInventoryCommand: user command when user want to see inventory
+     * addItemCommand: user command when user want to add item
+     * removeItemCommand: user command when user want to remove item
+     * useItemCommand: user command when user want to use item
+     * quitCommand: user command when user want to quit inventory
+     */
+    private Inventory inventory;
+    private boolean isDone = false;
+    private boolean awaitingInput = false;
+    private final String viewInventoryCommand;
+    private final String addItemCommand;
+    private final String removeItemCommand;
+    private final String useItemCommand;
+    private final String quitCommand;
+
+    /**
+     * @param inventory inventory contains items
+     */
+    public InventoryState(Inventory inventory){
+
+        this.inventory = inventory;
+        this.quitCommand = "quit";
+        this.viewInventoryCommand = "view";
+        this.removeItemCommand = "remove";
+        this.addItemCommand = "add";
+        this.useItemCommand = "use";
+
+    }
+
+    /**
+     * Executes when not awaiting input
+     */
+    @Override
+    public void preInput() {
+        awaitingInput = true;
+        // use outputHandler to display choices of commands
+    }
+
+    /**
+     * Executes when awaiting input from the user
+     * @param input from the user
+     *        Input with item should be like "use LEVEL 10 POTION"
+     *        Input without items - "view", "quit"
+     */
+    @Override
+    public void postInput(String input) {
+        boolean success = false;
+        if (input.length() == 4) {
+            if (input.equals("quit")){
```

Comment:
if you have 'quit' as a constant you don't need the private final variables
I suggest u use them